app/scripts/services/quota.js

```diff
@@ -227,36 +227,25 @@ angular.module("openshiftConsole")
     };
 
     var getLatestQuotaAlerts = function(resources, context) {
-      var result = $q.defer();
-      var quotaAlerts = [];
-      var quotas, clusterQuotas;
-      var remaining = 2;
-      function _checkDone() {
-        if (remaining === 0) {
-          quotaAlerts = getQuotaAlerts(resources, quotas, clusterQuotas);
-          result.resolve({quotaAlerts: quotaAlerts});
-        }
-      }
+      var quotas, clusterQuotas, promises = [];
 
       // double check using the latest quotas
-      DataService.list("resourcequotas", context, function(quotaData) {
+      promises.push(DataService.list("resourcequotas", context).then(function(quotaData) {
         quotas = quotaData.by("metadata.name");
         Logger.log("quotas", quotas);
-        remaining--;
-        _checkDone();
-      });
+      }));
 
-      DataService.list("appliedclusterresourcequotas", context, function(clusterQuotaData) {
+      promises.push(DataService.list("appliedclusterresourcequotas", context).then(function(clusterQuotaData) {
         clusterQuotas = clusterQuotaData.by("metadata.name");
         Logger.log("cluster quotas", clusterQuotas);
-        remaining--;
-        _checkDone();
-      });
+      }));
 
-      // TODO when DataService.List supports error callbacks an error retrieving the lists should reject the promise
-      // So that failing to check your quota doesn't block app creation.
-
-      return result.promise;
+      return $q.all(promises).then(function() {
+        var quotaAlerts = getQuotaAlerts(resources, quotas, clusterQuotas);
+        return {
+          quotaAlerts: quotaAlerts
+        };
+      });
     };
 
     var isAnyQuotaExceeded = function(quotas, clusterQuotas) {
```

Comment:
i feel like it needs a comment added to explicitly say that it will not trigger on a quota of zero